### PR TITLE
Support latest version of SAM

### PR DESCRIPTION
**Pull request by silvexis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,40 +23,36 @@ support you there and back again.
...
     $ sam deploy --capabilities CAPABILITY_IAM --region us-east-1 --stack-name my-cool-stack --parameter-overrides Namespace=dev
         ...
-    $ aws cloudformation describe-stack-events --stack-name my-cool-stack --profile my-account
-        ...
 
-There are a few other items that complicate matters like not being able to do simple variable substitution in
-a CloudFormation template, MFA support is poorly thought out and there is no way to extend the build system
-(e.g. plugins).
+There are a few other items that complicate matters like not being able to do simple variables in
+a CloudFormation template (I'm sorry, but mappings are just plain ugly), MFA support is poorly thought out
```
Comment (Tankanow) [inline README.md]: IMHO, I'm fine with the MFA support ... I assume a role outside of the tool and it honors it.

File: README.md [hunk 2]
```diff
@@ -23,40 +23,36 @@ support you there and back again.
...
+(requires multiple prompts and doesn't cache!) and there is no way to extend the build system (e.g. plugins).
```
Comment (Tankanow) [inline README.md]: IMHO, I am not a fan of plugin systems for build frameworks. It just ends up being messy.

File: samwise/__main__.py [hunk 3]
```diff
@@ -72,32 +72,41 @@ def main():
             display(parsed_template_obj)
 
     elif arguments.get('package'):
-        aws_creds = get_aws_credentials(aws_profile)
-        pre_process_template(metadata, output_location, template_obj)
-        package(output_location, base_dir, aws_creds, s3_bucket, parameter_overrides)
+        # aws_creds = get_aws_credentials(aws_profile)
```
Comment (Tankanow) [inline samwise/__main__.py]: SHOULD: Remove commented out line.

File: samwise/__main__.py [hunk 4]
```diff
@@ -72,32 +72,41 @@ def main():
...
+        aws_creds = None
+        parsed_template_obj = pre_process_template(metadata, output_location, template_obj)
+        result = package.create(parsed_template_obj, output_location, base_dir, aws_creds, s3_bucket)
+        print(result)
+        # sam_package(output_location, base_dir, aws_creds, s3_bucket, parameter_overrides)
```
Comment (Tankanow) [inline samwise/__main__.py]: SHOULD: Remove `print` and commented out line

File: samwise/__main__.py [hunk 5]
```diff
@@ -72,32 +72,41 @@ def main():
...
+        sam_package(output_location, base_dir, aws_creds, s3_bucket, parameter_overrides)
         deploy(aws_creds, aws_profile, deploy_region, output_location, stack_name, namespace, parameter_overrides)
     else:
         print('Nothing to do')
 
 
 def deploy(aws_creds, aws_profile, deploy_region, output_location, stack_name, namespace, parameter_overrides=None):
     print(f" - Deploying Stack '{namespace}-{stack_name}' to AWS profile '{aws_profile}'")
-    command = ["aws", "cloudformation", "deploy",
+    # command = ["aws", "cloudformation", "deploy",
```
Comment (Tankanow) [inline samwise/__main__.py]: SHOULD: Remove commented out lines

File: samwise/features/package.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2019 CloudZero, Inc. All rights reserved.
+# Licensed under the MIT License. See LICENSE file in the project root for full license information.
+
+import docker
+from nested_lookup import nested_lookup
+
+
+def check_docker(runtimes):
+    client = docker.from_env()
+    print(client.images.list())
```
Comment (Tankanow) [inline samwise/features/package.py]: ?: Do you want these `print` statements in this module?

File: samwise/features/package.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    python_runtimes = nested_lookup('Runtime', parsed_template_obj)
+    print(python_runtimes)
+
+    return python_runtimes
+
+
+def create(parsed_template_obj, output_location, base_dir, aws_creds, s3_bucket):
+    python_runtimes = get_python_runtimes(parsed_template_obj)
+
+    result = check_docker(python_runtimes)
```
Comment (Tankanow) [inline samwise/features/package.py]: I think this `result` is a docker client. Is this how the python `docker` module works? It looks like lines 35-37 are just a string. Is that run somehow?

File: samwise/features/template.py [hunk 8]
```diff
@@ -58,3 +60,33 @@ def save(template_yaml_obj, output_file_location):
...
+def parse(template_obj, metadata):
+    processed_variables = {}
+    variables = metadata.get(constants.VARS_KEY, [])
+
+    for var in variables:
+        if not isinstance(var, dict):
+            value = input(f" - {var} : ")
+            processed_variables[var] = value
+        else:
+            processed_variables.update(dict(var))
```
Comment (Tankanow) [inline samwise/features/template.py]: `var` must be a `dict` here, no need to cast it again

File: samwise/utils/cli.py [hunk 9]
```diff
@@ -1,35 +1,46 @@
...
 import sys
-from subprocess import PIPE, Popen
+from select import select
+from subprocess import Popen
 
 
-def execute_and_process(command, transform=None, env=None):
-    os.environ['PYTHONUNBUFFERED'] = "1"
+def execute_and_process(command, env=None):
+    """Largely found in https://stackoverflow.com/a/31953436"""
```
Comment (Tankanow) [inline samwise/utils/cli.py]: This seems like a lot just to get stdout/stderr processed together, when it's 2 characters in bash `&>`. 😉 

Seems fine I think.

Comment (silvexis) [reply samwise/utils/cli.py]: you have no idea. Things get crazy when you want the output to stream to the screen and support ANSI colors

File: README.md (hunk 1 above)
Comment (silvexis) [reply README.md]: yes but you have to use a tool outside of the tool, just to use the tool, that's bad

File: README.md (hunk 2 above)
Comment (silvexis) [reply README.md]: meh?

**Pull request merged**